### ISSUE TITLE
MDBOOK001 should support explicit plain text code blocks

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#163]
## Description
MDBOOK001 reports warnings for code blocks missing language tags even when those blocks are intentionally plain text (command output, help text, etc.).

## Current Behavior
44 instances of warnings like:
```
cli-reference/profile/README.md:3:1: warning: MDBOOK001/code-block-language: Code block is missing language tag
```

## Use Cases for Plain Text Blocks
- CLI help output
- Command execution results
- Log files
- Configuration file templates
- Plain text data

## Feature Request
Consider supporting explicit "no highlighting" markers to suppress this warning:
- ```text
- ```plain
- ```none
- ```output

This would differentiate between:
1. Accidentally missing language tags (should warn)
2. Intentionally plain text blocks (shouldn't warn)

## Example
```markdown
# This should warn (missing language)
\`\`\`
some code here
\`\`\`

# This shouldn't warn (explicitly plain)
\`\`\`text
Usage: redisctl [OPTIONS] [COMMAND]

Commands:
  profile  Manage connection profiles
  help     Print this message
\`\`\`
```

## Environment
- mdbook-lint: v0.11.2
- OS: macOS Darwin 24.6.0

## Test Repository
https://github.com/joshrotenberg/redisctl (branch: fix/reenable-mdbook-lint-issue-37)
